daliuge-engine/dlg/apps/pyfunc.py

```diff
@@ -80,22 +82,29 @@ def serialize_func(f):
 
 
 def import_using_name(app, fname):
-    # The name has the form pack1.pack2.mod.func
+    # If only one part check if builtin
     parts = fname.split(".")
     if len(parts) < 2:
-        msg = "%s does not contain a module name" % fname
-        raise InvalidDropException(app, msg)
-
-    modname, fname = ".".join(parts[:-1]), parts[-1]
-    try:
-        mod = importlib.import_module(modname, __name__)
-        return getattr(mod, fname)
-    except ImportError as e:
-        raise InvalidDropException(
-            app, "Error when loading module %s: %s" % (modname, str(e))
-        )
-    except AttributeError:
-        raise InvalidDropException(app, "Module %s has no member %s" % (modname, fname))
+        b = globals()['__builtins__']
+        logger.debug(f"Builtins: {type(b)}")
+        logger.debug(f"Function {fname}: {hasattr(b, fname)}")
```

Comment:
This is doing string interpolation before the logging statement is filtered in/out. These particular log statements might not cause too much trouble because they are tiny and not in a tight loop, but in general it's best to avoid this and go with `logging`'s way of interpolating log messages. Let's go with these changes as they are though, and I'll create a tech debt ticket to do a full sweep of this in the daliuge codebase (I recently did some of this in 3068a2f8 as they were potentially affecting the performance of the translator).

Reply:
I've created https://jira.skatelescope.org/browse/YAN-981 to keep track of this.